components/Snackbar/tests/unit/MDCSnackbarColorThemerTests.m

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialSnackbar+ColorThemer.h"
+#import "MaterialSnackbar.h"
+
+@interface MDCSnackbarColorThemerTests : XCTestCase
+
+@property(nonatomic, strong) UIColor *messageTextColor;
+@property(nonatomic, strong) UIColor *snackbarMessageViewShadowColor;
+@property(nonatomic, strong) UIColor *snackbarMessageViewBackgroundColor;
+@property(nonatomic, strong) NSMutableDictionary *titleColorForState;
+
+@end
+
+@implementation MDCSnackbarColorThemerTests
+
+- (void)setUp {
+  [super setUp];
+
+  self.messageTextColor = MDCSnackbarManager.messageTextColor;
+  self.snackbarMessageViewShadowColor = MDCSnackbarManager.snackbarMessageViewShadowColor;
+  self.snackbarMessageViewBackgroundColor = MDCSnackbarManager.snackbarMessageViewBackgroundColor;
+  self.titleColorForState = [@{} mutableCopy];
+  NSUInteger maxState = UIControlStateNormal | UIControlStateDisabled | UIControlStateSelected |
+                        UIControlStateHighlighted;
+  for (NSUInteger state = 0; state < maxState; ++state) {
+    self.titleColorForState[@(state)] = [MDCSnackbarManager buttonTitleColorForState:state];
+  }
+}
+
+- (void)tearDown {
+  // Restore the Snackbar Manager's state
+  MDCSnackbarManager.messageTextColor = self.messageTextColor;
+  MDCSnackbarManager.snackbarMessageViewShadowColor = self.snackbarMessageViewShadowColor;
+  MDCSnackbarManager.snackbarMessageViewBackgroundColor = self.snackbarMessageViewBackgroundColor;
+  for (NSNumber *state in self.titleColorForState.allKeys) {
+    if (self.titleColorForState[state] != nil) {
+      [MDCSnackbarManager setButtonTitleColor:self.titleColorForState[state]
+                                     forState:state.unsignedIntegerValue];
+    }
+  }
+
+  // Clean-up the test case
+  [self.titleColorForState removeAllObjects];
+  self.messageTextColor = nil;
+  self.snackbarMessageViewShadowColor = nil;
+  self.snackbarMessageViewBackgroundColor = nil;
+
+  [super tearDown];
+}
+
+- (void)testSnackbarColorThemerChangesCorrectParameters {
+  // Given
+  MDCSemanticColorScheme *colorScheme = [[MDCSemanticColorScheme alloc] init];
+  colorScheme.surfaceColor = [UIColor redColor];
+  colorScheme.onSurfaceColor = [UIColor blueColor];
+  MDCSnackbarManager.snackbarMessageViewBackgroundColor = [UIColor whiteColor];
```

Comment:
Please use an instance of the manager instead of the singleton.

Reply:
See https://github.com/material-components/material-components-ios/blob/develop/components/Snackbar/tests/unit/MDCSnackbarManagerInstanceTests.m for examples, though we should use properties now instead of reallocating in every test.